### ISSUE TITLE
Caching nil and failing to reach BING

### ISSUE THREAD

**Issue by luigi7up** [alexreisner/geocoder#872]
Hi guys, first of all thanks for this great GEM. It's areal gem!

I am using it to get the coordinates for the address provided like , for example, Geocoder.coordinates("Toledo, Spain") and I get an array ["lat_val","long_val"]. When everything goes well it's great. Often it doesn't. Some of the problems are probably Bing maps problem and not the GEM's, but I'll mention them to see if anyone has some workarounds

SETUP:
initializer:

```
  lookup: :bing,
  key: ENV['BING_GEOCODE_ID'],
  api_key: ENV['BING_GEOCODE_ID'],
  cache: Rails.cache,
```

* Using Heroku with memcachier

I am experiencing these problems:
1) I get nil as a result quite often. For example:
```Geocoder.coordinates("San Francisco, CA") #nil```
I have noticed that downcasing and removing commas from the city name helps
```Geocoder.coordinates("San Francisco CA") # OK ["xxx","yyy"]```
This is probably the Bing maps problem although if I do this in their map tool it always works

2) I tried ```calling Geocoder.coordinates("San Francisco, CA")``` few times and clearing the cache after each.  If I get "Bing Geocoding API error: Service Unavailable" in my first call *nil* gets cached as a result so all subsequent calls are returned nil (from cache). 

Questions:
Would setting the timeout help avoiding the "Bing Geocoding API error: Service Unavailable"  result?
Is there a way of saying to Geocoder cache to ignore results that are not an array with values in it?

**Comment by alexreisner** [comment]
Thanks!

(1) It's a Bing issue and I don't think there's anything that can be done about it except to learn what kinds of queries the geocoding API "likes." Note that getting coordinates for something as broad as "San Francisco" is perhaps a little dubious anyway. Also, yes, the Bing website (as well as Google Maps) are using their own geocoding APIs in conjunction with other services (which they don't provide access to) so you're always going to get slightly different results.

(2) This was fixed by #847. If you install from the current HEAD it should work. A new gem version (including that fix) will also be released very soon.

**Comment by luigi7up** [comment]
Alex,

thanks so much! I am bundling the master branch as we speak to fix the caching of nil issue.

I guess I'll try Google as provider since Bing is failing to often ;)

Thanks again!
Luka
